library/index.c

```diff
@@ -412,10 +460,49 @@ static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
 			break;
 		}
 		sorted--;
+	    } else {
+		/* read next block of headers */
+		if (!hdr_block_offset)
+			break;
+
+		if (hdr_block_offset < (uint64_t)hdr_block_size) {
+			hdr_block_size = hdr_block_offset;
+			num = hdr_block_size / hdr_size;
+			hdr_block_offset = 0;
+		} else {
+			hdr_block_offset -= hdr_block_size;
+		}
+
+		read_err = __eblob_read_ll(bctl->sort.fd, hdr_block, hdr_block_size, hdr_block_offset);
+		if (read_err < 0) {
+			err = read_err;
+			break;
+		}
+
+		/*
+		 * if it is first and initial block read then start searching record starting
+		 * from record preceded to founded by bsearch record (sorted_orig),
+		 * otherwise starting from last block record
+		 */
+		if (hdr_block_offset == saved_hdr_block_offset) {
+			sorted = sorted_orig - 1;
```

Comment:
Can sorted_orig - 1 be an invalid pointer, i.e. sorted_orig points to the very first header?